### DISCUSSION TITLE
how to deal with recurring date ranges where the year shouldn't be present?

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1256] [Q&A (Help)] [answered]
I'd like to add an option that makes it so a dialog is shown every year between an adjustable date range. eg. Jan to Apr 1 or Jan 15 to Apr 15 or whatever.

Pursuant to that, the date field having a year seems kinda weird.

[date_picker](https://backpackforlaravel.com/docs/6.x/crud-fields#date_picker-pro) seems like the most customizable date dialog but it doesn't look like even that one will let you select yearless date ranges?

What are your thoughts on how to deal with this?

[a comment by an automated account is not shown]

**Comment by jcastroa87** [comment, marked as answer]
Hi @prescriptionlifeline 

You're right that removing the year from a date field can make things tricky when using a calendar picker. Calendar views rely on the year to correctly determine the day of the week, which makes them less practical for selecting yearless date ranges like "Jan 15 to Apr 15."

The best way to handle this use case would be to create a custom field. Instead of using a calendar picker, the custom field could provide two text inputs for entering the start and end of the range in a MM-DD format. This approach avoids the confusion of irrelevant calendar years and focuses solely on capturing the day and month values. You can then add backend validations to ensure the inputs are in the correct format and handle the logic to compare dates without considering the year.

Cheers.

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
